### PR TITLE
(BIDS-2361) make the ValidatorSync function prettier

### PR DESCRIPTION
**Pull request by MauserBitfly**
### <samp>🤖 Generated by Copilot at 851a319</samp>

Refactor `ValidatorSync` handler to use maps, contexts, and errgroups for sync committee duties, and remove unused function `GetValidatorSyncDutiesHistoryOrdered` from `db` package.

### CODE REVIEW

File: handlers/validator.go [hunk 1]
```diff
@@ -1882,190 +1879,117 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
...
+							mux.Unlock()
+							continue // this is not the slot we are looking for
+						}
+
+						// We want to know how many validators successfully participated in this duty
+						if duty.Status == 1 {
+							pageSlot.Participation++
+						}
+
+						// Get the status if the dutiy belongs to our validator
```
Comment (Eisei24) [inline handlers/validator.go:1966]: **typo:** duty

File: handlers/validator.go [hunk 2]
```diff
@@ -1882,190 +1879,117 @@ func ValidatorSync(w http.ResponseWriter, r *http.Request) {
...
+		if descOrdering {
+			utils.ReverseSlice(syncSlots)
+		}
+
+		// let's pick the slots we want to display on the requested page
+		pageSlots := syncSlots[start:protomath.MinU64(start+length, totalCount)]
+		length = uint64(len(pageSlots)) // Last page might be shorter
+		epochs := []uint64{}
+		last := uint64(0)
+		// we gether all epochs for that page (the table currently displays 10 items per page, so it displays max 2 epoches. But we want to be future proof, so we support bigger page sizes)
```
Comment (Eisei24) [inline handlers/validator.go:1913]: **typo:** gather

**Review by Eisei24 — APPROVED** [review]
lgtm

This function now looks much more readable.
One potential issue is that we now call `GetValidatorSyncDutiesHistory` (which checks the bigtable) once for each epoch that is displayed on the current page as opposed to before where we had an epoch range as long as we were in the same sync committee period.

As a result these pages (every 4th page) take a bit longer to load.
@recy21 Please consider this when doing your check.

File: handlers/validator.go (hunk 1 above)
Comment (MauserBitfly) [reply handlers/validator.go]: fixed

File: handlers/validator.go (hunk 2 above)
Comment (MauserBitfly) [reply handlers/validator.go]: fixed

**Comment by MauserBitfly** [comment]
> lgtm
> 
> This function now looks much more readable. One potential issue is that we now call `GetValidatorSyncDutiesHistory` (which checks the bigtable) once for each epoch that is displayed on the current page as opposed to before where we had an epoch range as long as we were in the same sync committee period.
> 
> As a result these pages (every 4th page) take a bit longer to load. @recy21 Please consider this when doing your check.

Thanks. Yes every 4th page is a little slower than the other 3 (because it needs to load 2 epochs), but still faster then the old way ;-).

**Review by peterbitfly — APPROVED** [review]
LGTM DB wise, did not check the business logic

**Pull request merged**